frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -1100,6 +1221,39 @@ pub mod pallet {
             }
         }
 
+        /// Calculate rewards for all stakers for this era
+        fn stakers_era_reward(
+            points: EraStakingPoints<T::AccountId, BalanceOf<T>>,
+            reward_for_contract: BalanceOf<T>,
+        ) -> Vec<(T::AccountId, BalanceOf<T>)> {
+            let staker_part = Perbill::from_rational(reward_for_contract, points.total);
+            let mut res: Vec<(T::AccountId, BalanceOf<T>)> = Vec::new();
+
+            for (s, b) in &points.stakers {
+                let staker_reward = staker_part * *b;
+                res.push((s.clone(), staker_reward));
+            }
+            res
+        }
+
+        /// Summarize accumulated rewards and execute payout for stakers
+        fn payout_stakers2(rewards: Vec<Vec<(T::AccountId, BalanceOf<T>)>>) {
+            let mut staker_map: BTreeMap<T::AccountId, BalanceOf<T>> = Default::default();
+
+            // first collect all rewards from different eras
+            for era_rewards in rewards {
+                for (s, b) in era_rewards {
+                    let mut reward = staker_map.entry(s).or_insert(Default::default());
+                    *reward += b;
+                }
+            }
```

Comment:
Already commented before but since it also relates to this code - you can avoid double vecs altogether and just use map directly.